echopype/tests/consolidate/test_consolidate.py

```diff
@@ -1,6 +1,162 @@
+        range_kwargs,
+    )
+
+
+def _check_swap(ds, ds_swap):
+    assert "channel" in ds.dims
+    assert "frequency_nominal" not in ds.dims
+    assert "frequency_nominal" in ds_swap.dims
+    assert "channel" not in ds_swap.dims
+
+
+def test_swap_dims_channel_frequency(test_data_samples):
+    """
+    Test swapping dimension/coordinate from channel to frequency_nominal.
+    """
+    (
+        filepath,
+        sonar_model,
+        azfp_xml_path,
+        range_kwargs,
+    ) = test_data_samples
+    ed = ep.open_raw(filepath, sonar_model, azfp_xml_path)
+    if ed.sonar_model.lower() == 'azfp':
+        avg_temperature = (
+            ed['Environment']['temperature'].mean('time1').values
+        )
+        env_params = {
+            'temperature': avg_temperature,
+            'salinity': 27.9,
+            'pressure': 59,
+        }
+        range_kwargs['env_params'] = env_params
+        if 'azfp_cal_type' in range_kwargs:
+            range_kwargs.pop('azfp_cal_type')
+
+    dup_freq_valueerror = (
+        "Duplicated transducer nominal frequencies exist in the file. "
+        "Operation is not valid."
+    )
+
+    Sv = ep.calibrate.compute_Sv(ed, **range_kwargs)
+    try:
+        Sv_swapped = ep.consolidate.swap_dims_channel_frequency(Sv)
+        _check_swap(Sv, Sv_swapped)
+    except Exception as e:
+        assert isinstance(e, ValueError) is True
+        assert str(e) == dup_freq_valueerror
+
+    MVBS = ep.preprocess.compute_MVBS(Sv)
+    try:
+        MVBS_swapped = ep.consolidate.swap_dims_channel_frequency(MVBS)
+        _check_swap(Sv, MVBS_swapped)
+    except Exception as e:
+        assert isinstance(e, ValueError) is True
+        assert str(e) == dup_freq_valueerror
+
+
+def _build_ds_Sv(channel, range_sample, ping_time, sample_interval):
+    return xr.Dataset(
+        data_vars={
+            "Sv": ( 
+                ("channel", "range_sample", "ping_time"),
+                np.random.random((len(channel), range_sample.size, ping_time.size)),
+            ),
+            "echo_range": (
+                ("channel", "range_sample", "ping_time"),
+                (
+                    np.swapaxes(np.tile(range_sample, (len(channel), ping_time.size, 1)), 1, 2)
+                    * sample_interval
+                ),
+            ),
+        },
+        coords={
+            "channel": channel,
+            "range_sample": range_sample,
+            "ping_time": ping_time,
+        },
+    )
+
+
+def test_add_depth():
+    # Build test Sv dataset
+    channel = ["channel_0", "channel_1", "channel_2"]
+    range_sample = np.arange(100)
+    ping_time = pd.date_range(start="2022-08-10T10:00:00", end="2022-08-10T12:00:00", periods=121)
+    sample_interval = 0.01
+    ds_Sv = _build_ds_Sv(channel, range_sample, ping_time, sample_interval)
+
+    # no water_level in ds
+    try:
+        ds_Sv_depth = ep.consolidate.add_depth(ds_Sv)
+    except ValueError:
+        ...
+
+    # user input water_level
```

Comment:
Just to be more explicit, maybe add a comment:
```
# Use defaults `vertical=True` and `downward=True`

Reply:
I removed the option `vertical`